specification/intro.adoc

```diff
@@ -1,4 +1,86 @@
+:imagesdir: ./images
+
 [[intro]]
 
 == Introduction
 
+The RISC-V Confidential VM Extension <<CoVE>> specification provides
+application and virtualized workloads with data confidentiality and integrity,
+addressing one of the major datacenter security challenge. By building a
+miminized Trusted Computing Base (TCB), the CoVE interfaces manage to isolate
+workloads from each others but also from untrusted domain software components
+like the host hypervisor. CoVE implementations create a
+Confidential Computing <<CC>> framework that allows for mitigating the risks that
+guest owners are exposed to when running their workloads on shared
+infrastructures like e.g. public clouds.
+
+The value of such hardarwe-based Trusted Execution Environments (TEEs) is
+acknowledged across the industry, and Confidential Computing providers are
+looking for ways to improve the technology's scalability and performance. With
+the growing need for securing data processing, expanding Confidential Computing
+guests TCBs with trusted devices is of particular interest. Data analytics and
+transformation, artificial intelligence, financial transactions processing are
+only a few examples of confidential workloads for which a secure and performant
+I/O architecture is key to their operations. By extending guests TCBs with
+trusted accelerators, NICs or GPUs, such workloads would fully take advantage of
+their infrastructure providers capacities while keeping their data protected.
+
+The CoVE interfaces provide TEE Virtual Machines (TVMs) memory with confidential
+attributes and allow TVM guests to share parts of their address space with an
+untrusted domain. Although this enables confidential guests to be assigned with
+devices, directly or not, through para-virtualized I/O implementation, it comes
+with a significant performance cost. Withouth additional protection, TVMs have
+no ways to trust host-assigned devices and must exclude them from their TCB by
+not allowing them to directly access confidential memory. Consequently, with the
+current CoVE specification, data flowing from a device to a TVM must first go
+through dedicated shared memory regions for the confidential guest to then move
+it over to its confidential address space. This systematic data copy between
+shared and confidential memory is called bounce-buffering and can have a
+major performance impact for confidential workloads.
+
+[[bounce_buffer]]
+.Bounce buffering between an untrusted device and a CoVE TVM
+image::bounce_buffer.svg[]
+
+In addition to the performance degradation aspects, using a shared memory buffer
+for the TVM and the device to communicate brings security concerns as well. As
```

Comment:
I feel the word - "brings security concerns" may lead some confusing, because in the following section, we have described the mitigation - "this could be mitigated through device specific protocols (e.g. transport-level data encryption".
Then actually, there is no security concern since we already have solution.

I suggest we can just say "using a shared memory buffer requires additional protection of the communication between the device and TVM (e.g. transport-level data encryption) "

Reply:
That makes sense. I tried to address that comment, let me know if that's not the case.